### ISSUE TITLE
[Bug] Missing DNS name in the server certificate

### ISSUE THREAD

**Issue by jpkrohling** [strimzi/strimzi-kafka-operator#2027]
**Describe the bug**
The TLS certificate provided by the broker to clients is tied to the pod it's running and does not provide the statefulset's name (nor service name):

```
$ kafka-console-consumer -topic jaeger-spans -brokers my-cluster-kafka-brokers.kafka.svc.cluster.local:9093 -tls-client-cert /tmp/user.crt -tls-client-key /tmp/user.key -tls-enabled -verbose
2019/09/27 12:03:47 Initializing new client
2019/09/27 12:03:47 ClientID is the default of 'sarama', you should consider setting it to something application-specific.
2019/09/27 12:03:47 ClientID is the default of 'sarama', you should consider setting it to something application-specific.
2019/09/27 12:03:47 client/metadata fetching metadata for all topics from broker my-cluster-kafka-brokers.kafka.svc.cluster.local:9093
2019/09/27 12:03:47 Failed to connect to broker my-cluster-kafka-brokers.kafka.svc.cluster.local:9093: x509: certificate is valid for my-cluster-kafka-0.my-cluster-kafka-brokers.kafka.svc.cluster.local, my-cluster-kafka-bootstrap.kafka.svc, my-cluster-kafka-bootstrap.kafka.svc.cluster.local, my-cluster-kafka-bootstrap, my-cluster-kafka-bootstrap.kafka, not my-cluster-kafka-brokers.kafka.svc.cluster.local
2019/09/27 12:03:47 client/metadata got error from broker -1 while fetching metadata: x509: certificate is valid for my-cluster-kafka-0.my-cluster-kafka-brokers.kafka.svc.cluster.local, my-cluster-kafka-bootstrap.kafka.svc, my-cluster-kafka-bootstrap.kafka.svc.cluster.local, my-cluster-kafka-bootstrap, my-cluster-kafka-bootstrap.kafka, not my-cluster-kafka-brokers.kafka.svc.cluster.local
2019/09/27 12:03:47 client/metadata no available broker to send metadata request to
2019/09/27 12:03:47 client/brokers resurrecting 1 dead seed brokers
2019/09/27 12:03:47 client/metadata retrying after 250ms... (3 attempts remaining)
2019/09/27 12:03:47 ClientID is the default of 'sarama', you should consider setting it to something application-specific.
2019/09/27 12:03:47 client/metadata fetching metadata for all topics from broker my-cluster-kafka-brokers.kafka.svc.cluster.local:9093
2019/09/27 12:03:47 Failed to connect to broker my-cluster-kafka-brokers.kafka.svc.cluster.local:9093: x509: certificate is valid for my-cluster-kafka-0.my-cluster-kafka-brokers.kafka.svc.cluster.local, my-cluster-kafka-bootstrap.kafka.svc, my-cluster-kafka-bootstrap.kafka.svc.cluster.local, my-cluster-kafka-bootstrap, my-cluster-kafka-bootstrap.kafka, not my-cluster-kafka-brokers.kafka.svc.cluster.local
2019/09/27 12:03:47 client/metadata got error from broker -1 while fetching metadata: x509: certificate is valid for my-cluster-kafka-0.my-cluster-kafka-brokers.kafka.svc.cluster.local, my-cluster-kafka-bootstrap.kafka.svc, my-cluster-kafka-bootstrap.kafka.svc.cluster.local, my-cluster-kafka-bootstrap, my-cluster-kafka-bootstrap.kafka, not my-cluster-kafka-brokers.kafka.svc.cluster.local
2019/09/27 12:03:47 client/metadata no available broker to send metadata request to
2019/09/27 12:03:47 client/brokers resurrecting 1 dead seed brokers
2019/09/27 12:03:47 client/metadata retrying after 250ms... (2 attempts remaining)
2019/09/27 12:03:47 ClientID is the default of 'sarama', you should consider setting it to something application-specific.
2019/09/27 12:03:47 client/metadata fetching metadata for all topics from broker my-cluster-kafka-brokers.kafka.svc.cluster.local:9093
2019/09/27 12:03:47 Failed to connect to broker my-cluster-kafka-brokers.kafka.svc.cluster.local:9093: x509: certificate is valid for my-cluster-kafka-0.my-cluster-kafka-brokers.kafka.svc.cluster.local, my-cluster-kafka-bootstrap.kafka.svc, my-cluster-kafka-bootstrap.kafka.svc.cluster.local, my-cluster-kafka-bootstrap, my-cluster-kafka-bootstrap.kafka, not my-cluster-kafka-brokers.kafka.svc.cluster.local
2019/09/27 12:03:47 client/metadata got error from broker -1 while fetching metadata: x509: certificate is valid for my-cluster-kafka-0.my-cluster-kafka-brokers.kafka.svc.cluster.local, my-cluster-kafka-bootstrap.kafka.svc, my-cluster-kafka-bootstrap.kafka.svc.cluster.local, my-cluster-kafka-bootstrap, my-cluster-kafka-bootstrap.kafka, not my-cluster-kafka-brokers.kafka.svc.cluster.local
2019/09/27 12:03:47 client/metadata no available broker to send metadata request to
2019/09/27 12:03:47 client/brokers resurrecting 1 dead seed brokers
2019/09/27 12:03:47 client/metadata retrying after 250ms... (1 attempts remaining)
2019/09/27 12:03:48 ClientID is the default of 'sarama', you should consider setting it to something application-specific.
2019/09/27 12:03:48 client/metadata fetching metadata for all topics from broker my-cluster-kafka-brokers.kafka.svc.cluster.local:9093
2019/09/27 12:03:48 Failed to connect to broker my-cluster-kafka-brokers.kafka.svc.cluster.local:9093: x509: certificate is valid for my-cluster-kafka-0.my-cluster-kafka-brokers.kafka.svc.cluster.local, my-cluster-kafka-bootstrap.kafka.svc, my-cluster-kafka-bootstrap.kafka.svc.cluster.local, my-cluster-kafka-bootstrap, my-cluster-kafka-bootstrap.kafka, not my-cluster-kafka-brokers.kafka.svc.cluster.local
2019/09/27 12:03:48 client/metadata got error from broker -1 while fetching metadata: x509: certificate is valid for my-cluster-kafka-0.my-cluster-kafka-brokers.kafka.svc.cluster.local, my-cluster-kafka-bootstrap.kafka.svc, my-cluster-kafka-bootstrap.kafka.svc.cluster.local, my-cluster-kafka-bootstrap, my-cluster-kafka-bootstrap.kafka, not my-cluster-kafka-brokers.kafka.svc.cluster.local
2019/09/27 12:03:48 client/metadata no available broker to send metadata request to
2019/09/27 12:03:48 client/brokers resurrecting 1 dead seed brokers
2019/09/27 12:03:48 Closing Client
ERROR: Failed to start consumer: kafka: client has run out of available brokers to talk to (Is your cluster reachable?)
```

**To Reproduce**
1. Install the operator into minikube: 
    ```console
    $ kubectl create namespace kafka
    $ curl -L https://github.com/strimzi/strimzi-kafka-operator/releases/download/0.13.0/strimzi-cluster-operator-0.13.0.yaml | sed 's/namespace: .*/namespace: kafka/' | kubectl apply -f - -n kafka
    ```
1. Create Custom Resource:
    ```console
    $ kubectl apply -f https://raw.githubusercontent.com/strimzi/strimzi-kafka-operator/0.13.0/examples/kafka/kafka-persistent-single.yaml -n kafka 
    $ kubectl wait kafka/my-cluster --for=condition=Ready --timeout=300s -n kafka
    ```
1. Create a `KafkaUser`:
    ```console
    $ kubectl apply -n kafka -f - <<EOF
    apiVersion: kafka.strimzi.io/v1beta1
    kind: KafkaUser
    metadata:
      name: my-user
      labels:
        strimzi.io/cluster: my-cluster
    spec:
      authentication:
        type: tls
    EOF
    ```
1. Extract the secrets:
    ```console
    $ for secret in ca.crt user.crt user.key; do kubectl get secrets -n kafka my-user -o go-template="{{index .data \"${secret}\"}}" | base64 -d > /tmp/${secret} ; done
    ```
1. Expose the service as a local port
    ```console
    $ kubectl port-forward svc/my-cluster-kafka-brokers -n kafka 9093
    ```
1. Add the hostname to `/etc/hosts`
    ```console
    echo "127.0.0.1 my-cluster-kafka-brokers.kafka.svc.cluster.local" | sudo tee -a /etc/hosts
    ```
1.  Try to get a client connected to the brokers:
    ```console
    $ kafka-console-consumer -topic jaeger-spans -brokers my-cluster-kafka-brokers.kafka.svc.cluster.local:9093 -tls-client-cert /tmp/user.crt -tls-client-key /tmp/user.key -tls-enabled -verbose
    ```

The mentioned error is then shown:
```
2019/09/27 12:03:48 Failed to connect to broker my-cluster-kafka-brokers.kafka.svc.cluster.local:9093: x509: certificate is valid for my-cluster-kafka-0.my-cluster-kafka-brokers.kafka.svc.cluster.local, my-cluster-kafka-bootstrap.kafka.svc, my-cluster-kafka-bootstrap.kafka.svc.cluster.local, my-cluster-kafka-bootstrap, my-cluster-kafka-bootstrap.kafka, not my-cluster-kafka-brokers.kafka.svc.cluster.local
```

**Expected behavior**
The connection is established, or, when the CA isn't trusted, the following would be shown when running the same `kafka-console-consumer` command:
```
2019/09/27 12:07:22 client/metadata fetching metadata for all topics from broker my-cluster-kafka-0.my-cluster-kafka-brokers.kafka.svc.cluster.local:9093
2019/09/27 12:07:23 Failed to connect to broker my-cluster-kafka-0.my-cluster-kafka-brokers.kafka.svc.cluster.local:9093: x509: certificate signed by unknown authority
2019/09/27 12:07:23 client/metadata got error from broker -1 while fetching metadata: x509: certificate signed by unknown authority
```

**Environment (please complete the following information):**
 - Strimzi version: [e.g. master, 0.13.0]
 - Installation method: YAML files
 - Kubernetes cluster: Kubernetes 1.15
 - Infrastructure: Minikube

**Additional context**
Found while testing jaegertracing/jaeger#1414

**Comment by jpkrohling** [comment]
As shared via other channels, looks like the appropriate service is `my-cluster-kafka-bootstrap`, and not `my-cluster-kafka-brokers`. I find it a bit counter-intuitive, and there could be a better documentation about it, but I agree that this is not a bug. I'm therefore closing this.

**Comment by tombentley** [comment]
@jpkrohling please could you point us to the documentation you would inadequate?

**Comment by jpkrohling** [comment]
@tombentley perhaps there could be a paragraph under the section [2.4.1. Deploying the Kafka cluster](https://strimzi.io/docs/latest/#deploying-kafka-cluster-str), like:

> Once your Kafka cluster is ready, it will be reachable via the service name `${instance-name}-bootstrap`.

**Comment by tombentley** [comment]
@jpkrohling I agree, this isn't adequately explained. [2.8. Deploying example clients](https://strimzi.io/docs/latest/#deploying-example-clients-str), explains it in the context of running an example client, and at the very least we should add a link to 2.8 from the "See also" section of 2.4.1. @PaulRMellor could you fix this? Let me know if you need a separate issue for it.

**Comment by scholzj** [comment]
Just for the record - the PR #2070 added the names of the headless services as well.
